presto-main/src/main/java/com/facebook/presto/execution/FailedQueryExecution.java

```diff
@@ -45,16 +48,37 @@
     private final QueryInfo queryInfo;
     private final Session session;
     private final Executor executor;
-    private final Optional<ResourceGroupId> resourceGroup;
 
-    public FailedQueryExecution(QueryId queryId, String query, Optional<ResourceGroupId> resourceGroup, Session session, URI self, TransactionManager transactionManager, Executor executor, Metadata metadata, Throwable cause)
+    public static FailedQueryExecution createFailedQueryExecution(
+            String query,
+            Session session,
+            URI self,
+            TransactionManager transactionManager,
+            Executor executor,
+            Metadata metadata,
+            Throwable cause)
+    {
+        QUERY_STATE_LOG.debug(cause, "Query %s failed", session.getQueryId());
```

Comment:
I've never seen a public logger shared between classes. Not sure why we went in this direction.